deepforest/visualize.py

```diff
@@ -257,37 +257,225 @@ def label_to_color(label):
+
+        # Extract labels as a numpy array
+        labels = df['label'].map(label_mapping).values.astype(int)
 
-    label_mapping = {label: idx for idx, label in enumerate(df['label'].unique())}
+        # Extract scores as a numpy array
+        scores = np.array(df['score'].tolist())
+        scores = np.expand_dims(np.stack(scores), 1)
 
-    # Extract labels as a numpy array
-    labels = df['label'].map(label_mapping).values.astype(int)
+        # Create a reverse mapping from integer to string labels
+        class_name = {v: k for k, v in label_mapping.items()}
 
-    # Extract scores as a numpy array
-    scores = np.array(df['score'].tolist())
-    # Create a reverse mapping from integer to string labels
-    class_name = {v: k for k, v in label_mapping.items()}
+        detections = sv.KeyPoints(
+            xy=points,
+            class_id=labels,
+            confidence=scores,
+            data={"class_name": [class_name[class_id] for class_id in labels]})
 
-    return sv.Detections(
-        xyxy=boxes,
-        class_id=labels,
-        confidence=scores,
-        data={"class_name": [class_name[class_id] for class_id in labels]})
+    return detections
+
+
+def plot_results(results,
+                 ground_truth=None,
+                 savedir=None,
+                 height=None,
+                 width=None,
+                 results_color=[245, 135, 66],
+                 ground_truth_color=[0, 165, 255],
+                 thickness=2,
+                 basename=None,
+                 radius=3):
+    """Plot the prediction results.
+
+    Args:
+        df: a pandas dataframe with prediction results
+        ground_truth: an optional pandas dataframe with ground truth annotations
+        savedir: optional path to save the figure. If None (default), the figure will be interactively plotted.
+        height: height of the image in pixels. Required if the geometry type is 'polygon'.
+        width: width of the image in pixels. Required if the geometry type is 'polygon'.
+        basename: optional basename for the saved figure. If None (default), the basename will be extracted from the image path.
+        results_color (list): color of the results annotations as a tuple of RGB color, e.g. orange annotations is [245, 135, 66]
+    Returns:
+        None
+    """
+    # Convert colors, check for multi-class labels
+    results_color_sv = sv.Color(results_color[0], results_color[1], results_color[2])
+    ground_truth_color_sv = sv.Color(ground_truth_color[0], ground_truth_color[1],
+                                     ground_truth_color[2])
+
+    num_labels = len(results.label.unique())
+    if num_labels > 1 and results_color is not None:
+        warnings.warn(
+            "Multiple labels detected in the results. Each label will be plotted with a different color using a color ramp, results color argument is ignored."
+        )
+    if num_labels > 1:
+        sv_color = sv.ColorPalette.from_matplotlib('viridis', 5)
```

Comment:
A few thoughts here:

1. `results_color` will never be `None` since it's set to a color in the default now. One option would be to change this to:

```python
    if num_labels > 1 and len(results_color) != num_labels:
        warnings.warn(
            "Multiple labels detected in the results. Each label will be plotted with a different color using a color ramp, results color argument is ignored."
        )
        if num_labels > 1:
            sv_color = sv.ColorPalette.from_matplotlib('viridis', 5)
```

This would allow the users to pass in their own color ramp of the appropriate length (we could also use `<` just to make sure it's long enough) and then it would be used (might need a check to see if it is an `sv.ColorPalette`). In fact it looks like you've anticipated this in the docs, but anything passed in would be overridden in the current implementation (I think). If the data is multi-class but only has a single color (or fewer colors than they need) they'll get the warning and the default viridis color ramp will be used.

2. Looks like this is hard coding at 5 color classes. Should we set this dynamically based on the number of classes? A quick local test shows it's happy to return an arbitrarily large number of colors
3. Also, should this be overwriting `results_color_sv` instead of create `sv_color`? That seems to be what is being passed to _plot_image_with_results below